browser/brave_rewards/checkout_dialog.cc

```diff
@@ -0,0 +1,317 @@
+    rewards_service_ = RewardsServiceFactory::GetForProfile(profile);
+  }
+  return rewards_service_;
+}
+
+void CheckoutDialogHandler::RegisterMessages() {
+  web_ui()->RegisterMessageCallback(
+      "paymentRequestComplete",
+      base::BindRepeating(&CheckoutDialogHandler::HandlePaymentCompletion,
+                          base::Unretained(this)));
+  web_ui()->RegisterMessageCallback(
+      "getWalletBalance",
+      base::BindRepeating(&CheckoutDialogHandler::OnGetWalletBalance,
+                          base::Unretained(this)));
+  web_ui()->RegisterMessageCallback(
+      "getExternalWallet",
+      base::BindRepeating(&CheckoutDialogHandler::GetExternalWallet,
+                          base::Unretained(this)));
+  web_ui()->RegisterMessageCallback(
+      "getRewardsParameters",
+      base::BindRepeating(&CheckoutDialogHandler::GetRewardsParameters,
+                          base::Unretained(this)));
+}
+
+void CheckoutDialogHandler::HandlePaymentCompletion(
+    const base::ListValue* args) {
+  auto spec = request_->spec();
+  if (!request_->spec()->IsInitialized()) {
+    return;
+  }
+
+  auto* rfh =
+      content::RenderFrameHost::FromID(request_->initiator_frame_routing_id());
+  if (!rfh) {
+    request_->OnError(PaymentErrorReason::INVALID_DATA_FROM_RENDERER,
+                      errors::kInvalidRenderer);
+    return;
+  }
+
+  auto* rewards_service = GetRewardsService();
+  if (!rewards_service) {
+    request_->OnError(PaymentErrorReason::INVALID_DATA_FROM_RENDERER,
+                      errors::kRewardsNotInitialized);
+    return;
+  }
+
+  const auto& display_items =
+      spec->GetDisplayItems(request_->state()->selected_app());
+  for (size_t i = 0; i < display_items.size(); i++) {
+    DCHECK((*display_items[i])->sku.has_value());
+    auto item = ledger::type::SKUOrderItem::New();
+
+    item->sku = (*display_items[i])->sku.value();
+    item->quantity = 1;
+    items_.push_back(std::move(item));
+  }
+
+  auto callback = base::BindOnce(&OnSKUProcessed, request_);
+
+  rewards_service->ProcessSKU(
+      std::move(items_), ledger::constant::kWalletUphold, std::move(callback));
+}
+
+void CheckoutDialogHandler::GetRewardsParameters(const base::ListValue* args) {
+  if (auto* service = GetRewardsService()) {
+    AllowJavascript();
+    service->GetRewardsParameters(
+        base::Bind(&CheckoutDialogHandler::GetRewardsParametersCallback,
+                   weak_factory_.GetWeakPtr()));
+  }
+}
+
+void CheckoutDialogHandler::GetRewardsParametersCallback(
+    ledger::type::RewardsParametersPtr parameters) {
+  if (!IsJavascriptAllowed()) {
+    return;
+  }
+
+  base::Value data(base::Value::Type::DICTIONARY);
+  if (parameters) {
+    data.SetDoubleKey("rate", parameters->rate);
+    data.SetDoubleKey("lastUpdated", base::Time::Now().ToJsTimeIgnoringNull());
+  }
+  FireWebUIListener("rewardsParametersUpdated", data);
+}
+
+void CheckoutDialogHandler::OnGetWalletBalance(const base::ListValue* args) {
```

Comment:
why is this method called `OnGetWalletBalance` if the method is `GetWalletBalance`?

Reply:
resolved